src/Model.jl

```diff
@@ -45,19 +43,27 @@ struct Model{T <: Real}
     symops::Vector{SymOp}
 end
 
+"""
+Model(lattice; kwargs...)
+
+Creates the physical specification of a model (without any
+discretization information).
+
+The `symmetry` kwarg can be:
+auto: determine from terms if they respect the symmetry of the lattice/atoms.
+off: no symmetries at all
+force: force all the symmetries of the lattice/atoms.
+Careful that in this last case, wrong results can occur if the
+external potential breaks symmetries (this is not checked).
+"""
 function Model(lattice::AbstractMatrix{T};
-               n_electrons=nothing,
+               n_electrons=nothing,  # taken from `atoms` if not specified
                atoms=[],
                terms=[],
                temperature=0.0,
-               smearing=nothing,
-               spin_polarization=:none, # ∈ (:none, :collinear, :full, :spinless)
-               symmetry=:auto # auto: determine from terms if they are symmetric.
-                              # off: no symmetries at all
-                              # force: force all the symmetries of the lattice/atoms.
-                              # Careful that in the forcing case, wrong results
-                              # can occur if the external potential breaks symmetries
-                              # (this is not checked)
+               smearing=nothing,  # By default: Fermi-Dirac if temperature != 0, none otherwise
+               spin_polarization=:none,  # ∈ (:none, :collinear, :full, :spinless)
```

Comment:
I really disagree with the comments here. It should be in the docstring or in the struct.

Reply:
Hm I'm not too sure. These don't belong in the struct because they concern default arguments to the constructor. I really don't like putting it in the constructor docstring because it quickly turns into a "`atoms` is the list of atoms, `terms` is the list of terms, temperature is the temperature" type thing and doesn't really add more information than reading the code. This kind of quick remarks is helpful to understand the default values for anybody reading the code.

Reply:
Then just explains the ones you would remark in the comment. I think that is a good compromise.